src/frontends/lean/definition_cmds.cpp

```diff
@@ -249,6 +249,9 @@ declare_definition(parser & p, environment const & env, decl_cmd_kind kind, buff
     } else {
         c_real_name = get_namespace(env) + c_name;
     }
+    if (env.find(c_real_name)) {
+        throw exception(sstream() << "invalid definition, a declaration named '" << c_real_name << "' has already been declared");
+    }
     if (val && !meta.m_modifiers.m_is_meta && !type_checker(env).is_prop(type)) {
```

Comment:
Great! I hate that weird kernel exception too.

Reply:
The other kernel exception I see regularly is "forgot `meta`", but that wasn't a simple 3-lines fix.